### PR TITLE
Critical qca_spi fixes and some improvements

### PR DESCRIPTION
**Pull request by lategoodbye**
This pull request contains some critical fixes and some minor improvements on the qca_spi driver.

### CODE REVIEW

File: drivers/net/ethernet/qualcomm/qca_debug.c
Comment (mooraby) [inline drivers/net/ethernet/qualcomm/qca_debug.c:1]: is the _ in the commit title ignored intentionally? 'qca_spi'

File: drivers/net/ethernet/qualcomm/qca_spi.h [hunk 1]
```diff
@@ -41,6 +41,7 @@
 
 #define QCASPI_TX_RING_MAX_LEN 10
 #define QCASPI_TX_RING_MIN_LEN 2
+#define QCASPI_RX_MAX_FRAMES 4
```
Comment (mooraby) [inline drivers/net/ethernet/qualcomm/qca_spi.h:31]: maybe I am mixing stuff, but regarding the naming, is this related to RX ring paramters? And is this related to this commit https://github.com/chargebyte/linux/pull/3/commits/65a92c8162f9451d9a995d5b880be8ac9632b6e5?

File: drivers/net/ethernet/qualcomm/qca_spi.c [hunk 2]
```diff
@@ -581,6 +581,11 @@ qcaspi_spi_thread(void *data)
 	netdev_info(qca->net_dev, "SPI thread created\n");
```
Comment (mooraby) [inline drivers/net/ethernet/qualcomm/qca_spi.c:568]: Just couple of stuff regarding the commit message:

- "This is a big issue because it allows userspace to prevent from restarting the SPI thread..." So the issue is that the userspace is prevented from restarting the thread or that the userspace itself is not allowing the thread to restart?
- "The open/close ops could be realized just by park/unpark" should be parking/unparking.. but I see that you are referring the kernel thread functions so maybe but it in single quotes

File: drivers/net/ethernet/qualcomm/qca_spi.c [hunk 3]
```diff
@@ -810,8 +810,8 @@ qcaspi_netdev_init(struct net_device *dev)
 	qca->clkspeed = qcaspi_clkspeed;
```
Comment (mooraby) [inline drivers/net/ethernet/qualcomm/qca_spi.c:797]: 1. mtu is a mutual parameter, but actually* we need the maximum possible

File: drivers/net/ethernet/qualcomm/qca_7k.c [hunk 4]
```diff
@@ -1,22 +1,7 @@
+// SPDX-License-Identifier: GPL-2.0 OR BSD-2-Clause
```
Comment (mooraby) [inline drivers/net/ethernet/qualcomm/qca_7k.c:1]: In some changes the multiple line comment style /* */ is used and in other the single line //..

Comment (lategoodbye) [reply drivers/net/ethernet/qualcomm/qca_7k.c:1]: This is some strang camp fire knowledge about the SPDX tag. In C files this should be // and in header files it should /* . AFAIK this is because of some kind of tool incompatibility.

File: drivers/net/ethernet/qualcomm/qca_debug.c
Comment (lategoodbye) [reply drivers/net/ethernet/qualcomm/qca_debug.c:1]: I'm not sure which specific commit you refering to? [ac96d77](https://github.com/chargebyte/linux/pull/3/commits/ac96d7790afd10e26e6396c34fd273fbeaa8c64d) Yes this unintended and must be fixed.

File: drivers/net/ethernet/qualcomm/qca_spi.h (hunk 1 above)
Comment (lategoodbye) [reply drivers/net/ethernet/qualcomm/qca_spi.h:31]: The problem is there is no such a thing as RX ring. Yes, this value is the same thing, but this is the reason it has been dropped from ethtool -G support.

File: drivers/net/ethernet/qualcomm/qca_spi.c (hunk 2 above)
Comment (lategoodbye) [reply drivers/net/ethernet/qualcomm/qca_spi.c:568]: 1. The problem is that these functions can be called by user context. In case the user hit Strg+C the signal HUP would be send to the SPI thread of the driver which is unintended.
2. Yes, i'm refering to the kernel thread functions.

File: drivers/net/ethernet/qualcomm/qca_spi.c (hunk 3 above)
Comment (lategoodbye) [reply drivers/net/ethernet/qualcomm/qca_spi.c:797]: Do you refer to the commit message [e030d2c](https://github.com/chargebyte/linux/pull/3/commits/e030d2c70a5267f7aecd4b3bdf2a81f50bc518fa)

File: drivers/net/ethernet/qualcomm/qca_debug.c
Comment (mooraby) [reply drivers/net/ethernet/qualcomm/qca_debug.c:1]: yes that's the commit I meant

File: drivers/net/ethernet/qualcomm/qca_spi.c (hunk 3 above)
Comment (mooraby) [reply drivers/net/ethernet/qualcomm/qca_spi.c:797]: yes

File: drivers/net/ethernet/qualcomm/qca_debug.c
Comment (lategoodbye) [reply drivers/net/ethernet/qualcomm/qca_debug.c:1]: Fixed

File: drivers/net/ethernet/qualcomm/qca_spi.c (hunk 2 above)
Comment (lategoodbye) [reply drivers/net/ethernet/qualcomm/qca_spi.c:568]: Please have a look

File: drivers/net/ethernet/qualcomm/qca_spi.c (hunk 3 above)
Comment (lategoodbye) [reply drivers/net/ethernet/qualcomm/qca_spi.c:797]: Fixed

File: drivers/net/ethernet/qualcomm/qca_spi.c (hunk 2 above)
Comment (mooraby) [reply drivers/net/ethernet/qualcomm/qca_spi.c:568]: looks good

**Review by mooraby — APPROVED** [review]

**Comment by mooraby** [comment]
Tested with the scripts, but did not do a full test e.g. a charing session

**Review by mhei — APPROVED** [review]
LGTM

**Comment by mhei** [comment]
I don't delete the branch for now, but @lategoodbye please clean it up once you don't need it anymore.
